### PR TITLE
tests: detect stray dbus-daemon

### PR DESCRIPTION
**Pull request by zyga**
The DBus specification contains the unfortunate instructions on how to
launch a new DBus session bus if one is not already availalbe. All of
our tests should now run with session-managed DBus and should not run a
DBus daemon themselves. To avoid weird bugs, detect the case where more
than one "dbus-dameon" process is running. Note that this check is
performed before preparation and after restoring, so well-behaved tests
should no longer have anything around.

Signed-off-by: Zygmunt Krynicki <me@zygoon.pl>

### CODE REVIEW

**Comment by zyga** [comment]
NOTE: This will fail as there are still tests that use dbus-launch and not clean up.

**Comment by zyga** [comment]
I've rebased this on master to check where we are. I think only one test will now fail the invariant but let's see what the system says.

**Comment by sergiocazzolato** [comment]
I see just those tests failing because of the invariant tool detection:
google:ubuntu-18.04-64:tests/main/lxd-no-fuse
google:ubuntu-18.04-64:tests/main/lxd-snapfuse

File: tests/lib/tools/invariant-tool [hunk 1]
```diff
@@ -74,6 +75,34 @@ check_lxcfs_mounted() {
 	fi
 }
 
+check_stray_dbus_daemon() {
+	n="$1" # invariant name
```
Comment (stolowski) [inline tests/lib/tools/invariant-tool:79]: Nitpick, could use a more descriptive var name.

Comment (zyga) [reply tests/lib/tools/invariant-tool:79]: I followed a pattern shared by other but I actually thinking about it, I can make invariant code easier to handle by removing that whole section and handling it once below. tl;dr; is that shared code, including this name, will go away. I'll send a follow-up.

Comment (stolowski) [reply tests/lib/tools/invariant-tool:79]: Sounds good, thank you.

**Pull request merged**